generic/tests/boot.py

```diff
@@ -18,15 +19,56 @@ def run(test, params, env):
     :param env: Dictionary with test environment.
     """
 
+    def get_valid_nic(vm):
+        """
+        return first nic which has an IP address assigned system
+        :param vm: vm object
+        :return: nic_index which is the int type(e.g. 0, 1, 2.) or None
+        """
+        nic_number_list = params.objects("nics")
+        for idx in range(len(nic_number_list)):
+            if vm.wait_for_get_address(idx):
+                return idx
+        return None
+
+    def multi_nics_reboot(session, vm):
+        """
+        Reboot vm which has multi nic devices and return session after login
+        NOTE: There is only nic, at random, which is assigned by dhcp
+              to an IP address after reboot. This nic should be found from
+              many nic devices and this nic will be used in login function.
+
+        :params vm: vm object
+        :return: A new shell session object.
+        """
+        if params["reboot_method"] == "shell":
+            session.sendline(params.get("reboot_command"))
+        else:
+            error_context.context("Unknown reboot method: %s"
```

Comment:
Does this mean we don't plan to support `system_reset`? I don't think it's advisable, because this change will affect some other test cases. And `vm.reboot` also supports `nic_index=x`.

Reply:
> Does this mean we don't plan to support `system_reset`? I don't think it's advisable, because this change will affect some other test cases. And `vm.reboot` also supports `nic_index=x`.

After I checked the code of `vm.reboot()`, `vm.reboot()` does NOT support the `system_reset`. That's why I code to support the only way `shell`.
However, you think the `system_reset` way should be supported at here. I will change code to support.